texar/torch/data/data/scalar_data_test.py

```diff
@@ -68,18 +86,23 @@ def _run_and_test(self, hparams, test_transform=False):
             else:
                 self.assertEqual(i, value)
             i += 1
-            if hparams["dataset"]["data_type"] == "int":
+            data_type = hparams["dataset"]["data_type"]
+            if data_type == "int":
                 self.assertTrue(isinstance(value, torch.Tensor))
                 self.assertEqual(value.dtype, torch.int32)
-            else:
+            elif data_type == "float":
                 self.assertTrue(isinstance(value, torch.Tensor))
                 self.assertEqual(value.dtype, torch.float32)
+            elif data_type == "bool":
+                self.assertTrue(isinstance(value, torch.Tensor))
```

Comment:
A bit unrelated, but we can use `assertIsInstance` here, and since this check exists in all branches, we can move it outside.

Reply:
52fca5f